Sources/Plasma/Apps/plClient/plClient_Win.cpp

```diff
@@ -0,0 +1,185 @@
+/*==LICENSE==*
+
+CyanWorlds.com Engine - MMOG client, server and tools
+Copyright (C) 2011  Cyan Worlds, Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+Additional permissions under GNU GPL version 3 section 7
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#include "hsWindows.h"
+#include "hsResMgr.h"
+#include "hsTimer.h"
+#include "plTimerCallbackManager.h"
+
+#include <Shlobj.h>
+#include <shellapi.h>
+
+#include "plClient.h"
+
+#include "pnFactory/plFactory.h"
+#include "pnNetCommon/plNetApp.h"
+#include "plProgressMgr/plProgressMgr.h"
+
+extern ITaskbarList3* gTaskbarList;
+static std::vector<HMODULE> fLoadedDLLs;
+
+void plClient::IResizeNativeDisplayDevice(int width, int height, bool windowed)
+{
+    uint32_t winStyle, winExStyle;
+    if (windowed) {
+        // WS_VISIBLE appears necessary to avoid leaving behind framebuffer junk when going from windowed to a smaller window
+        winStyle = WS_OVERLAPPED | WS_CAPTION | WS_SYSMENU | WS_MINIMIZEBOX | WS_VISIBLE;
+        winExStyle = WS_EX_APPWINDOW | WS_EX_WINDOWEDGE;
+    } else {
+        winStyle = WS_VISIBLE;
+        winExStyle = WS_EX_APPWINDOW;
+    }
+    SetWindowLong(fWindowHndl, GWL_STYLE, winStyle);
+    SetWindowLong(fWindowHndl, GWL_EXSTYLE, winExStyle);
```

Comment:
This might be a good opportunity to change this to `SetWindowLongPtr()` (64-bit aware)